pkg/virt-controller/watch/job.go

```diff
@@ -15,24 +16,27 @@ import (
 )
 
 func migrationJobSelector() kubeapi.ListOptions {
-	fieldSelector := fields.Everything()
-	labelSelector, err := labels.Parse(kvirtv1.DomainLabel)
+	fieldSelector := fields.ParseSelectorOrDie(
+		"status.phase!=" + string(v1.PodPending) +
+			",status.phase!=" + string(v1.PodRunning) +
+			",status.phase!=" + string(v1.PodUnknown))
+	labelSelector, err := labels.Parse(kvirtv1.DomainLabel + "," + kvirtv1.MigrationLabel)
 	if err != nil {
 		panic(err)
 	}
 	return kubeapi.ListOptions{FieldSelector: fieldSelector, LabelSelector: labelSelector}
 }
 
-func NewJobController(vmService services.VMService, recorder record.EventRecorder, restClient *rest.RESTClient) (cache.Store, *kubecli.Controller) {
+func NewJobController(vmService services.VMService, recorder record.EventRecorder, clientSet *kubernetes.Clientset, restClient *rest.RESTClient) (cache.Store, *kubecli.Controller) {
 	selector := migrationJobSelector()
-	lw := kubecli.NewListWatchFromClient(restClient, "jobs", kubeapi.NamespaceDefault, selector.FieldSelector, selector.LabelSelector)
+	lw := kubecli.NewListWatchFromClient(clientSet.CoreV1().RESTClient(), "pods", kubeapi.NamespaceDefault, selector.FieldSelector, selector.LabelSelector)
```

Comment:
Are you sure about the first parameter?  We should not be using a Rest Client for CoreV1 objects, and the URL is wrong for 3rd party.